pkg/controller/opentelemetrycollector/reconcile_service.go

```diff
@@ -38,15 +63,25 @@ func (r *ReconcileOpenTelemetryCollector) reconcileService(ctx context.Context)
 
 func service(ctx context.Context) *corev1.Service {
 	instance := ctx.Value(opentelemetry.ContextInstance).(*v1alpha1.OpenTelemetryCollector)
-	name := resourceName(instance.Name)
+	logger := ctx.Value(opentelemetry.ContextLogger).(logr.Logger)
 
-	labels := commonLabels(ctx)
-	labels["app.kubernetes.io/name"] = name
+	name := resourceName(instance.Name)
+	labels := serviceLabels(ctx, name)
 
 	// by coincidence, the selector is the same as the label, but note that the selector points to the deployment
 	// whereas 'labels' refers to the service
 	selector := labels
 
+	ports, err := ports(logger, instance.Spec.Config)
+	if err != nil {
+		logger.Error(err, "couldn't build the service for this instance")
```

Comment:
Shouldn't the instance name be used as context for the logged message?

Reply:
The logger already contains a field with the name/namespace set at the very beginning of the reconciliation logic.